dataproc_jupyter_plugin/controllers/composer.py

```diff
@@ -33,10 +33,13 @@ async def get(self):
                 )
                 environments = await client.list_environments()
                 response = []
-                for environment in environments:
-                    env = environment.dict()
-                    response.append(env)
-                self.finish(json.dumps(response))
+                if type(environments) == list:
+                    for environment in environments:
+                        env = environment.dict()
+                        response.append(env)
+                    self.finish(json.dumps(response))
+                else:
+                    self.finish(environments)
```

Comment:
This is more complicated than it needs to be.

We can drop the iteration entirely and just set a default parameter to the `json.dumps` call:

```py
self.finish(json.dumps(environments, default=lambda x: x.dict()))
```

Reply:
Resolved